proto/proto_crypt.c

```diff
@@ -330,6 +334,9 @@ proto_crypt_enc(uint8_t * ibuf, size_t len, uint8_t obuf[PCRYPT_ESZ],
 	/* Sanity-check the length. */
 	assert(len <= PCRYPT_MAXDSZ);
 
+	/* Copy the original (initialized) context. */
+	memcpy(&ctx, &k->ctx_init, sizeof(HMAC_SHA256_CTX));
```

Comment:
Can you move this down to just before the "Append an HMAC" section?  Otherwise we might load data into the cache and then evict it before it's used.

Reply:
Rebased as requsted.